grammar.js

```diff
@@ -22,60 +22,55 @@ module.exports = grammar({
 
     rules: {
         source_file: $ => seq(
-            repeat($.decl),
-            repeat($.stmt),
-        ),
-
-        decl: $ => choice(
-            $.module_decl,
-            $.export_decl,
-            $.global_decl,
-            $.option_decl,
-            $.const_decl,
-            $.redef_decl,
-            $.redef_enum_decl,
-            $.redef_record_decl,
-            $.type_decl,
-            $.func_decl,
-            $.preproc,
+            repeat($._stmt),
         ),
 
         module_decl: $ => seq('module', $.id, ';'),
-        export_decl: $ => seq('export', '{', repeat($.decl), '}'),
+        export_decl: $ => seq('export', '{', repeat($._stmt), '}'),
 
         // A change here over Zeek's parser: we make the combo of init class
         // and initializer jointly optional, instead of individually. Helps
         // avoid ambiguity.
-        global_decl: $ => seq('global', $.id, optional(seq(':', $.type)), optional($.initializer), optional($.attr_list), ';'),
+        var_decl: $ => seq(choice('global', 'local'), $.id, optional(seq(':', $.type)), optional($.initializer), optional($.attr_list), ';'),
```

Comment:
This change and the one you flagged above was mainly driven by the desire to quickly identify which declarations are visible in a scope while implementing a Zeek language server. For that I do not really care whether a declaration was introduced by a _`global` declaration_ or _`local` statement_ since their effect is similar (they introduce a new identifier into some scope).

If the desire is for this parser is less parsing more localized pieces of source code into a potentially invalid parse tree, but to also perform more global validation (e.g., checking whether a certain construct can appear in a context), I can certainly work around it with some added verbosity on my end. I am not sure whether such constraints will cause issues, but I will try. FWIW, enforcing such constraints while creating the parse tree as opposed to when creating an AST from the parse tree seems to be the norm in Bison parsers 🤷.